zilliqa/src/consensus.rs

```diff
@@ -1233,6 +1283,9 @@ impl Consensus {
             return Ok(false);
         }
 
+        txn.tx.validate_gas_limit()?;
+        txn.tx.validate_input_size()?;
```

Comment:
Can we add this in a different PR? #1003 already exists and I think txn validation can rely on existing code from REVM. Also as @rrw-zilliqa points out there are implications for changing what we accept for Scilla that we should dig into.

I think transaction validation is a separate work item to block verification?